src/ews_xml.erl

```diff
@@ -213,6 +276,45 @@ find_ns([A | Rest], Nss) ->
     [A|find_ns(Rest, Nss)];
 find_ns([], _) -> [].
 
+%% ----------------------------------------------------------------------------
+%% Get All Nss
+ns_tag({txt, _Content}, NssMap) ->
+    NssMap;
+ns_tag({{Ns, txt}, _Content}, NssMap) ->
+    %% Ugly hack for namespaced txts
+    ns_add_prefix(Ns, NssMap);
+ns_tag({{Ns, _Name}, Attributes, Children}, NssMap0) ->
+    NssMap1 = ns_add_prefix(Ns, NssMap0),
+    NssMap2 = ns_attributes(Attributes, NssMap1),
+    ns_children(Children, NssMap2);
+ns_tag({_Name, Attributes, Children}, NssMap0) ->
+    NssMap1 = ns_attributes(Attributes, NssMap0),
+    ns_children(Children, NssMap1).
+
+ns_attributes([{{NsN, _N}, {NsV, _V}} | Attrs], NssMap0) ->
+    NssMap1 = ns_add_prefix(NsN, NssMap0),
+    NssMap2 = ns_add_prefix(NsV, NssMap1),
+    ns_attributes(Attrs, NssMap2);
+ns_attributes([{{Ns, _N}, _Value} | Attrs], NssMap0) ->
+    NssMap1 = ns_add_prefix(Ns, NssMap0),
+    ns_attributes(Attrs, NssMap1);
+ns_attributes([{_Key, _Value} | Attrs], NssMap) ->
+    ns_attributes(Attrs, NssMap);
+ns_attributes([], NssMap) ->
+    NssMap.
+
+ns_children([C | Children], NssMap0) ->
+    NssMap1 = ns_tag(C, NssMap0),
+    ns_children(Children, NssMap1);
+ns_children([], NssMap) ->
+    NssMap.
+
+ns_add_prefix(Ns, NssMap) ->
+    case maps:is_key(Ns, NssMap) of
+        true -> NssMap;
+        false -> NssMap#{Ns => true}
+    end.
```

Comment:
I think just adding it in the map gives less code to read without being too terse.

```suggestion
    maps:put(Ns, true, NssMap).
```

Reply:
I am coding for the most common case here, that there are many many instances of Ns, like thousands if it is a huge request, but there are mostly all the same. So in order not to create a lot of new maps, as in most cases it will already be in the map, I have the test there.

Reply:
We have seen messages with in the order of 100k entries, so that would a lot of extra maps on the heap.

Reply:
To quote the doc: "If key Key already exists in map Map1, the old associated value is replaced by value Value. The function returns a new map Map2 containing the new association and the old associations in Map1."

Reply:
Alright, good to know. 👍